[rocm-smi-lib] build fails due to relocation error at linker stage

I'm getting this error when building rocm-smi-lib (in a chroot with aurutils). Any idea how to fix that?
```
==> Making package: rocm-smi-lib 5.0.2-1
==> Checking runtime dependencies...
==> Installing missing dependencies...
resolving dependencies...
looking for conflicting packages...

Packages (10) hsa-amd-aqlprofile-bin-5.1.0-1  hsakmt-roct-5.1.0-1  hwdata-0.357-1  kmod-29-2  libdrm-2.4.110-1  libpciaccess-0.16-2  numactl-2.0.14-2
              pciutils-3.7.0-2  rocm-device-libs-5.1.0-1  hsa-rocr-5.1.0-1

Total Installed Size:  11.51 MiB

:: Proceed with installation? [Y/n]
(10/10) checking keys in keyring                                                                [########################################################] 100%
(10/10) checking package integrity                                                              [########################################################] 100%
(10/10) loading package files                                                                   [########################################################] 100%
(10/10) checking for file conflicts                                                             [########################################################] 100%
:: Processing package changes...
( 1/10) installing numactl                                                                      [########################################################] 100%
( 2/10) installing hwdata                                                                       [########################################################] 100%
( 3/10) installing kmod                                                                         [########################################################] 100%
( 4/10) installing pciutils                                                                     [########################################################] 100%
( 5/10) installing libpciaccess                                                                 [########################################################] 100%
( 6/10) installing libdrm                                                                       [########################################################] 100%
( 7/10) installing hsakmt-roct                                                                  [########################################################] 100%
( 8/10) installing rocm-device-libs                                                             [########################################################] 100%
( 9/10) installing hsa-amd-aqlprofile-bin                                                       [########################################################] 100%
(10/10) installing hsa-rocr                                                                     [########################################################] 100%
==> Checking buildtime dependencies...
==> Installing missing dependencies...
resolving dependencies...
looking for conflicting packages...
warning: dependency cycle detected:
warning: harfbuzz will be installed before its freetype2 dependency

Packages (61) aom-3.3.0-1  cairo-1.17.6-1  dav1d-0.9.2-1  ffcall-2.4-1  fontconfig-2:2.13.96-1  freetype2-2.11.1-1  gd-2.3.3-3  giflib-5.2.1-2
              graphite-1:1.3.14-1  harfbuzz-4.2.0-1  harfbuzz-icu-4.2.0-1  hicolor-icon-theme-0.17-2  jsoncpp-1.9.5-1  lcms2-2.13.1-1  libavif-0.9.3-1
              libde265-1.0.8-2  libheif-1.12.0-3  libice-1.0.10-3  libjpeg-turbo-2.1.3-1  libnsl-2.0.0-2  libpaper-1.1.28-1  libpng-1.6.37-3
              libsigsegv-2.14-1  libsm-1.2.3-2  libsynctex-2021.59745-2  libtiff-4.3.0-2  libuv-1.43.0-1  libwebp-1.2.2-1  libx11-1.7.3.1-1  libxau-1.0.9-3
              libxaw-1.0.14-1  libxcb-1.14-1  libxdmcp-1.1.3-3  libxext-1.3.4-3  libxmu-1.1.3-2  libxpm-3.5.13-2  libxrender-0.9.10-4  libxt-1.2.1-1
              libyuv-r2266+eb6e7bb6-1  lzo-2.10-3  nspr-4.33-1  nss-3.76.1-1  openjpeg2-2.4.0-1  perl-file-which-1.27-1  pixman-0.40.0-1  poppler-22.03.0-1
              potrace-1.16-2  rav1e-0.4.1-2  rhash-1.4.2-1  run-parts-5.5-1  svt-av1-0.9.0-2  t1lib-5.1.2-8  texlive-bin-2021.59745-2
              texlive-core-2021.62793-1  x265-3.5-3  xcb-proto-1.14.1-5  xorgproto-2021.5-1  zziplib-0.13.72-1  cmake-3.22.3-1  doxygen-1.9.3-1
              texlive-latexextra-2021.62806-1

Total Installed Size:  718.87 MiB

[snip...]

/usr/bin/ld: /tmp/ccoRYcwL.ltrans1.ltrans.o: warning: relocation against `_ZNSt17_Function_handlerIFbcENSt8__detail11_AnyMatcherINSt7__cxx1112regex_traitsIcEELb1ELb0ELb0EEEE10_M_managerERSt9_Any_dataRKS8_St18_Manager_operation' in read-only section `.text'
/usr/bin/ld: /tmp/ccoRYcwL.ltrans0.ltrans.o: relocation R_X86_64_PC32 against symbol `_ZZNSt19_Sp_make_shared_tag5_S_tiEvE5__tag' can not be used when making a shared object; recompile with -fPIC
/usr/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
make[2]: *** [oam/CMakeFiles/oam.dir/build.make:276: oam/liboam.so.1.0] Error 1
make[2]: Leaving directory '/build/rocm-smi-lib/src/build'
make[1]: *** [CMakeFiles/Makefile2:312: oam/CMakeFiles/oam.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

--- a/rocm-smi-lib/.SRCINFO
+++ b/rocm-smi-lib/.SRCINFO
@@ -1,6 +1,6 @@
 pkgbase = rocm-smi-lib
 	pkgdesc = ROCm System Management Interface Library
-	pkgver = 5.0.2
+	pkgver = 5.1.0
 	pkgrel = 1
 	url = https://github.com/RadeonOpenCompute/rocm_smi_lib
 	arch = x86_64
@@ -9,10 +9,11 @@ pkgbase = rocm-smi-lib
 	makedepends = doxygen
 	makedepends = texlive-latexextra
 	depends = hsa-rocr
-	provides = rocm-smi-lib64=5.0.2
+	provides = rocm-smi-lib64=5.1.0
 	conflicts = rocm-smi-lib64
 	replaces = rocm-smi-lib64
-	source = rocm-smi-lib-5.0.2.tar.gz::https://github.com/RadeonOpenCompute/rocm_smi_lib/archive/rocm-5.0.2.tar.gz
-	sha256sums = a169129e4ecd1cca134039dc1bf91e1b3721768781abfae4ae61fad60a633472
+	options = !lto
+	source = rocm-smi-lib-5.1.0.tar.gz::https://github.com/RadeonOpenCompute/rocm_smi_lib/archive/rocm-5.1.0.tar.gz
+	sha256sums = 21b31b43015b77a9119cf4c1d4ff3864f9ef1f34e2a52a38f985a3f710dc5f87
 
 pkgname = rocm-smi-lib

--- a/rocm-smi-lib/PKGBUILD
+++ b/rocm-smi-lib/PKGBUILD
@@ -2,7 +2,7 @@
 # Contributor: acxz <akashpatel2008 at yahoo dot com>
 
 pkgname=rocm-smi-lib
-pkgver=5.0.2
+pkgver=5.1.0
 pkgrel=1
 pkgdesc='ROCm System Management Interface Library'
 arch=('x86_64')
@@ -14,7 +14,8 @@ replaces=('rocm-smi-lib64')
 conflicts=('rocm-smi-lib64')
 makedepends=('cmake' 'doxygen' 'texlive-latexextra')
 source=("$pkgname-$pkgver.tar.gz::https://github.com/RadeonOpenCompute/rocm_smi_lib/archive/rocm-$pkgver.tar.gz")
-sha256sums=('a169129e4ecd1cca134039dc1bf91e1b3721768781abfae4ae61fad60a633472')
+sha256sums=('21b31b43015b77a9119cf4c1d4ff3864f9ef1f34e2a52a38f985a3f710dc5f87')
+options=(!lto)
 _dirname="$(basename "$url")-$(basename "${source[0]}" .tar.gz)"
 
 

